lib/internal/validators.js

```diff
@@ -547,7 +547,37 @@ function validateLinkHeaderValue(hints) {
   );
 }
 
-module.exports = {
+/**
+ * @callback validateDataTypes -  This validator could be used for checking the provided value is either
+ * a provided type.
+ * @param {*} value
+ * @param {string} name
+ * @param {[string]} [types] - The multiple types to be validated. For eg: `['String', 'Buffer', 'Boolean'] -
+ *  All supported types are [ Array, StringArray, BooleanArray, AbortSignalArray, Boolean, Buffer, Dictionary,
+ *  Encoding, Function, Int32, Integer, Number, Object, OneOf, PlainFunction, Port, SignalName, String, Uint32,
+ *  Undefined, Union, AbortSignal, LinkHeaderValue, InternalField, DataTypes ]`
+ * @returns {asserts value is string}
+ */
+const validateDataTypes = (value, name, types) => {
+
+  const errors = [];
+  for (const type of types) {
+    try {
+      const validatorFunction = functionDefinitions[`validate${type}`];
+      if (!validatorFunction) throw new ERR_INVALID_ARG_TYPE(name, types, value);
+
+      validatorFunction(value, name);
+    } catch (err) {
+      errors.push(err);
+    }
+  }
+
+  if (errors.length === types.length) {
+    throw new ERR_INVALID_ARG_TYPE(name, types, value);
+  }
+};
```

Comment:
None of the current FS methods use this, introducing it would still make for an inconsistant behavior. `opendir` should support `Buffer` instances the same way with or without the `recursive` option.

Reply:
its removed. no new validation adding as per this change.